packages/core/src/storage/vectorStore/MongoDBAtlasVectorStore.ts

```diff
@@ -119,7 +121,7 @@ export class MongoDBAtlasVectorSearch
     const params: any = {
       queryVector: query.queryEmbedding,
       path: this.embeddingKey,
-      numCandidates: query.similarityTopK * 10,
+      numCandidates: this.numCandidates ?? query.similarityTopK * 10,
```

Comment:
why is using `query.similarityTopK` not sufficient? 
(I also wonder why it is `*10` here...)

Reply:
MongoDB Atlas using ANN (https://arxiv.org/abs/1603.09320) for the vector search. If you have a small number for the `similarityTopK`, the candiates they use for the lookup might be to small to include all relevant chunks.

Reply:
@pserrer1 got it. thanks a lot for the context. I see. So the upper bound `* 10` is incorrect for your use case

Here's the documentation:
<img width="815" alt="image" src="https://github.com/run-llama/LlamaIndexTS/assets/17126/e44c1754-cbeb-461e-a7d4-665b248746bd">

They recommend a max of 20, does this work for you:

```
limit: query.similarityTopK,
numCandidates: query.similarityTopK * 20
```

Reply:
Not sure if we find a solution that fits all. I updated the file with js docs